starter-code/src/functions-and-arrays.js

```diff
@@ -74,8 +175,32 @@ var wordsCount = [
   'disobedience',
   'matter'
 ];
+
+function howManyTimes(wordArr, wordToSearch) {
+  var count = 0;
+  if(wordArr.length === 0) {
+    return false;
+  }
+  for (var i = 0; i < wordArr.length; i++) {
+// wordArr.forEach (function (oneWord){
+     if (wordArr[i] === wordToSearch) {
+       // if (oneWord === wordToSearch) {
+          count++;
+        }
+      }
+  return count;
+// });
+}
+
+howManyTimes(words,'matter'); 
+
+
```

Comment:
Well done!

Your code is clean, not too long, and easily readable! Keep up the good work.

ANd here is the bonus response : 
```
function greatestProduct(m) {
  var horizontal, vertical, diagonal;
  var greatProduct = 0;

  for (var i=0, nRows = m.length; i < nRows - 1; i++) {
    for (var j=0, nCols = m[0].length; j < nCols - 1; j++) {
      // 4 horizontal elements
      if (j < nCols-3) {
        horizontal = m[i][j] * m[i][j+1]   * m[i][j+2]   * m[i][j+3];
      }

      // 4 vertical elements
      if (i < nRows-3) {
        vertical   = m[i][j] * m[i+1][j]   * m[i+2][j]   * m[i+3][j];
      }

      // 4 Diagonal elements
      if (i < nRows-3 && j < nCols-3) {
        diagonal   = m[i][j] * m[i+1][j+1] * m[i+2][j+2] * m[i+3][j+3];
      }

      greatProduct = Math.max(greatProduct, horizontal, vertical, diagonal);
    }
  }

  return greatProduct;
}
```